ipam/allocator.go

```diff
@@ -696,8 +700,12 @@ func (alloc *Allocator) actorLoop(actionChan <-chan func(), stopChan <-chan stru
 			// Retry things in case messages got lost between here and recipients
 			if alloc.awaitingConsensus {
 				alloc.propose()
-			} else {
-				alloc.tryPendingOps()
+			} else if alloc.havePendingOps() {
+				if alloc.ring.Empty() {
+					alloc.establishRing()
+				} else {
+					alloc.tryPendingOps()
+				}
```

Comment:
I updated my suggestion earlier, after realising the need for the `alloc.havePendingOps()` so we don't establish a ring unnecessarily.

How about
```
} else if alloc.havePendingOps() {
    alloc.establishRing()
    if !alloc.ring.Empty() {
        alloc.tryPendingOps()
    }
}
```
?